Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1342,39 +1339,226 @@ namespace OpenInfraPlatform
+				* \param[out] targetVec				The tessellated line.
+				* \param[out] segmentStartPoints	The starting points of separate segments.
+				* \param[in] trim1Vec				The trimming of the curve as saved in IFC model - trim at start of curve.
+				* \param[in] trim2Vec				The trimming of the curve as saved in IFC model - trim at end of curve.
+				* \param[in] senseAgreement			Does the resulting geometry have the same sense agreement as the \c IfcOffsetCurve2D.
+				*
+				* \note The function is not implemented.
+				*/
+				void convertIfcOffsetCurve2D(
+					const EXPRESSReference<typename IfcEntityTypesT::IfcOffsetCurve2D>& offsetCurve2D,
+					std::vector<carve::geom::vector<3>>& targetVec,
+					std::vector<carve::geom::vector<3>>& segmentStartPoints,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim1Vec,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim2Vec,
+					const bool senseAgreement,
+					const typename IfcEntityTypesT::IfcTrimmingPreference& trimmingPreference
+				) const noexcept(false)
+				{
+					// **************************************************************************************************************************
+					//	https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/lexical/IfcOffsetCurve2D.htm
+					//	ENTITY IfcOffsetCurve2D
+					//		SUBTYPE OF(IfcOffsetCurve);
+					//			Distance: IfcLengthMeasure;
+					//			SelfIntersect: IfcLogical;
+					//		WHERE
+					//			DimIs2D : BasisCurve.Dim = 2;
+					//	END_ENTITY;
+					// **************************************************************************************************************************
+					EXPRESSReference<typename IfcEntityTypesT::IfcCurve> basisCurve = offsetCurve2D->BasisCurve;
+					std::vector<carve::geom::vector<3>> basisCurvePoints;
+					double distance = offsetCurve2D->Distance * this->UnitConvert()->getLengthInMeterFactor();
+
+					convertIfcCurve(basisCurve, basisCurvePoints, segmentStartPoints,
+						trim1Vec, trim2Vec, senseAgreement, trimmingPreference);
+
+					std::vector<carve::geom::vector<3>> offsetCurve2DPoints;
+
+					for (int i; i < basisCurvePoints.size() - 1; i++) {
+						carve::geom::vector<3> startPoint = basisCurvePoints[i];
+						carve::geom::vector<3> endPoint = basisCurvePoints[i + 1];
+
+						carve::geom::vector<2> startPoint2D = carve::geom::VECTOR(startPoint.x, startPoint.y);
+						carve::geom::vector<2> endPoint2D = carve::geom::VECTOR(endPoint.x, endPoint.y);
+
+						carve::geom::vector<2> tangent = endPoint2D - startPoint2D;
+						carve::geom::vector<2> orthogonal = carve::geom::VECTOR(-tangent.y, tangent.x);
+
+						carve::geom::vector<3> offsetPoint = carve::geom::VECTOR(startPoint.x + distance * orthogonal.x, 
+																				startPoint.y + distance * orthogonal.y, 
+																				startPoint.z);
+						offsetCurve2DPoints.push_back(offsetPoint);

```

Comment:
Going through the definition of IfcOffsetCurve2D it seems like any two dimensions could be provided.
So we would need to have different cases for XY, XZ, and YZ.
I guess the case covered by you with X and Y will be the most common one but we also should support the other ones.

Reply:
Resolved: [974d6b4](https://github.com/tumcms/Open-Infra-Platform/pull/558/commits/974d6b452edea3448da6796761d58c0401982b5c)